MINOR: Clean up TestUtils.scala

m1a2st opened the pull request:
Because TestUtils.scala has some unused method, and some methods only use in one class, so I delete unused methods and move methods to individual class which only one class used

m1a2st committed:
resolve conflict

m1a2st commented:
@chia7712 please review this PR, thank you

chia7712 commented:
@m1a2st Could you please fix the conflicts? thanks!

m1a2st committed:
resolve conflict

m1a2st commented:
@chia7712 I resolve the conflicts

chia7712 commented on core/src/test/scala/unit/kafka/utils/TestUtils.scala:
```diff
@@ -1594,10 +1313,6 @@ object TestUtils extends Logging {
     }
   }
 
-  def createIsrChangeListener(): MockAlterPartitionListener = {
```
`MockAlterPartitionListener` is used by `AbstractPartitionTest` only, so could you move it to `AbstractPartitionTest`?

chia7712 commented on core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala:
```diff
@@ -548,4 +549,31 @@ class DeleteTopicTest extends QuorumTestHarness {
+    try {
+      val newPartitionSet: Map[String, NewPartitions] = Map.apply(topic -> NewPartitions.increaseTo(totalPartitionCount))
+      admin.createPartitions(newPartitionSet.asJava)
+    } catch {
```
We do nothing for this catch, so it should be fine to remove it.

chia7712 commented on core/src/test/scala/unit/kafka/admin/DeleteTopicTest.scala:
```diff
@@ -548,4 +549,31 @@ class DeleteTopicTest extends QuorumTestHarness {
+
+      (0 until totalPartitionCount - 1).map { i =>
+        i -> allPartitionsMetadata.get(new TopicPartition(topic, i)).map(_.leader()).getOrElse(
+          throw new IllegalStateException(s"Cannot get the partition leader for topic: $topic, partition: $i in server metadata cache"))
```
this method return nothing, so we don't need to collect the metadata. However, it needs to check the existence of topic partition

chia7712 commented on core/src/test/scala/unit/kafka/log/LogSegmentTest.scala:
```diff
@@ -613,4 +613,19 @@ class LogSegmentTest {
+    file.seek(position)
+    for (_ <- 0 until size)
+      file.writeByte(random.nextInt(255))
+    file.close()
```
could you please add try-finally?

chia7712 commented on core/src/test/scala/integration/kafka/api/SaslPlainPlaintextConsumerTest.scala:
```diff
@@ -52,6 +56,20 @@ class SaslPlainPlaintextConsumerTest extends BaseConsumerTest with SaslSetup {
+   * the case when zookeeper.set.acl=false and no ACLs have been configured.
+   */
+  private def verifyUnsecureZkAcls(zkClient: KafkaZkClient): Unit = {
+    secureZkPaths(zkClient).foreach(path => {
```
We can merge this method into test case `testZkAclsDisabled`

chia7712 commented on core/src/test/scala/integration/kafka/api/SaslPlainSslEndToEndAuthorizationTest.scala:
```diff
@@ -149,6 +150,23 @@ class SaslPlainSslEndToEndAuthorizationTest extends SaslEndToEndAuthorizationTes
+  /**
+   * Verifies that all secure paths in ZK are created with the expected ACL.
+   */
+  private def verifySecureZkAcls(zkClient: KafkaZkClient, usersWithAccess: Int): Unit = {
```
ditto

chia7712 reviewed (commented):
@m1a2st thanks for nice cleanup! a couple of comments left. PTAL

m1a2st committed:
recording PR comment to refactor code

m1a2st commented:
@chia7712 thanls for your comment, I have already change these code by your comment

chia7712 commented:
@m1a2st Could you please fix conflicts?

chia7712 commented:
```
[2024-04-27T01:29:26.736Z] > Task :core:compileTestScala

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:20:33: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:20:40: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:25:25: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:31:17: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:34:40: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:34:56: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:53:49: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:58:46: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:58:62: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:58:86: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:58:109: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:59:33: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:60:30: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:63:40: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:63:60: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:66:43: Unused import

[2024-04-27T01:29:26.736Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala:68:54: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:61:33: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:61:55: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:63:54: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:63:74: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:63:92: Unused import

[2024-04-27T01:29:28.026Z] [Error] /home/jenkins/workspace/Kafka_kafka-pr_PR-15808/core/src/test/scala/unit/kafka/utils/TestUtils.scala:63:106: Unused import

[2024-04-27T01:29:29.239Z] 23 errors found

```

please fix the build error :)

m1a2st committed:
removed unused import

m1a2st committed:
removed unused import

m1a2st commented:
I have been remove these unused import

chia7712 commented on core/src/test/scala/unit/kafka/utils/TestUtils.scala:
```diff
@@ -1124,24 +933,18 @@ object TestUtils extends Logging {
-      value.isDefined
-    }, msg, waitTimeMs)
-    value.get
+  def subscribeAndWaitForRecords(topic: String,
```
why adding this new method? it is unused.

m1a2st replied:
`assertBadConfigContainingMessage` this method are used by two classes, so I didn't move it. 
The `kafka.KafkaConfigTest` and `server.KafkaConfigTest` are using it

m1a2st replied:
Other methods refactored by your comment.

chia7712 reviewed (commented):
@m1a2st thanks for updated PR. The following methods seems to be moved to callers also. Please help us do double-check.

1. `setReplicationThrottleForPartitions`
2. `removeReplicationThrottleForPartitions`
3. `removeReplicationThrottleForPartitions`
4. `stringifyTopicPartitions`
5. `assertBadConfigContainingMessage`
6. `makeLeaderForPartition`
7. `getMsgStrings`

m1a2st committed:
move setReplicationThrottleForPartitions, removeReplicationThrottleForPartitions, removeReplicationThrottleForPartitions to separate class
move removeReplicationThrottleForPartitions to ReassignPartitionsIntegrationTest
move makeLeaderForPartition to AdminZkClientTest
removed getMessageString method
removed subscribeAndWaitForRecords method

m1a2st committed:
move tempPropertiesFile to DynamicBrokerReconfigurationTest

chia7712 commented on core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala:
```diff
@@ -1086,4 +1087,10 @@ class RemoteIndexCacheTest {
     }
   }
+
+  private def numThreadsRunning(threadNamePrefix: String, isDaemon: Boolean): mutable.Set[Thread] = {
```
We can fix the build error by rewriting this function via java lambda. For example:
```scala
  private def numThreadsRunning(threadNamePrefix: String, isDaemon: Boolean): java.util.Set[Thread] = {
    Thread.getAllStackTraces.keySet.stream().filter { t =>
      isDaemon && t.isAlive && t.getName.startsWith(threadNamePrefix)
    }.collect(Collectors.toSet)
  }
```

m1a2st replied:
I change the function name, and modify using java set
```java  
private def getRunningCleanerThread: java.util.Set[Thread] = {
    Thread.getAllStackTraces.keySet()
      .stream()
      .filter(t => t.isAlive && t.getName.startsWith(REMOTE_LOG_INDEX_CACHE_CLEANER_THREAD))
      .collect(Collectors.toSet())
  }
  ```

chia7712 commented on core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala:
```diff
@@ -23,25 +23,26 @@ import org.apache.kafka.common.{TopicIdPartition, TopicPartition, Uuid}
 import org.slf4j.{Logger, LoggerFactory}
 
-import java.io.{File, FileInputStream, IOException, PrintWriter, UncheckedIOException}
+import java.io._
```
We should avoid using `_` since it could cause naming conflicts.

chia7712 reviewed (commented):
@m1a2st thanks for updated PR. two major comments left. Please take a look.

m1a2st committed:
wip

m1a2st committed:
wip

chia7712 commented:
@m1a2st please make sure the build gets work with scala 2.12. You can use following command: `./gradlew clean build -x test -PscalaVersion=2.12`

chia7712 commented on core/src/test/scala/unit/kafka/log/remote/RemoteIndexCacheTest.scala:
```diff
@@ -321,16 +322,16 @@ class RemoteIndexCacheTest {
+    var threads = getRunningCleanerThread
     assertEquals(1, threads.size,
-      s"Found unexpected ${threads.size} threads=${threads.map(t => t.getName).mkString(", ")}")
+      s"Found unexpected ${threads.size} threads=${threads.stream().map(t => t.getName).collect(Collectors.joining(", "))}")
```
`threads.stream().map[String](t => t.getName).collect(Collectors.joining(", "))`

You can define the `type` explicitly to fix the type mismatch in scala 2.12

m1a2st replied:
fix it, and run `./gradlew clean build -x test -PscalaVersion=2.12` is build successful

chia7712 replied:
nice job!

m1a2st committed:
wip

chia7712 reviewed (approved):
LGTM

```
./gradlew cleanTest :streams:test --tests EOSUncleanShutdownIntegrationTest.shouldWorkWithUncleanShutdownWipeOutStateStore :tools:test --tests MetadataQuorumCommandTest.testDescribeQuorumReplicationSuccessful --tests GetOffsetShellTest.testTopicPartitionsArgWithInternalIncluded :storage:test --tests TransactionsWithTieredStoreTest.testSendOffsetsWithGroupMetadata --tests TransactionsWithTieredStoreTest.testFailureToFenceEpoch --tests TransactionsWithTieredStoreTest.testSendOffsetsToTransactionTimeout :connect:runtime:test --tests org.apache.kafka.connect.integration.ExactlyOnceSourceIntegrationTest.testSeparateOffsetsTopic :metadata:test --tests QuorumControllerTest.testFenceMultipleBrokers :streams:examples:test --tests WordCountDemoTest.testOneWord --tests WordCountDemoTest.testCountListOfWords --tests WordCountDemoTest.testGetStreamsConfig :connect:mirror:test --tests IdentityReplicationIntegrationTest.testSyncTopicConfigs --tests IdentityReplicationIntegrationTest.testOffsetSyncsTopicsOnTarget --tests MirrorConnectorsIntegrationBaseTest.testReplicationWithEmptyPartition --tests MirrorConnectorsIntegrationBaseTest.testSyncTopicConfigs --tests MirrorConnectorsIntegrationTransactionsTest.testSyncTopicConfigs :core:test --tests AddPartitionsTest.testIncrementPartitions --tests ConnectionQuotasTest.testListenerConnectionRateLimitWhenActualRateAboveLimit --tests TransactionsTest.testFencingOnCommit :clients:test --tests KafkaConsumerTest.testShouldAttemptToRejoinGroupAfterSyncGroupFailed
```
loop them 3 times, and all pass. will merge this PR
